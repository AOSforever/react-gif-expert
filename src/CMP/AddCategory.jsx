import { useState } from "react"

export const AddCategory = ( { onNewCategory } ) => {
    const [inputValue, setInputValue] = useState('Search Gifs');
    const onChangeValue = ({target}) => setInputValue(target.value);
    const onSubmit = event => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;
        onNewCategory( inputValue.trim() );
        setInputValue('');
    };
  return (
    <form onSubmit={ onSubmit }>
        <input 
            type="text" 
            placeholder="Buscar Gifs"
            value={inputValue}
            onChange={ onChangeValue }
        />
    </form>
  )
}
