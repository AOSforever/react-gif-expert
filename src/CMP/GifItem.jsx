import React from 'react'

export const GifItem = ({title,url}) => {
  return (
    <div className='card'>
        <img src={`${url}/giphy.gif`} alt={title} />
    </div>
  )
}
