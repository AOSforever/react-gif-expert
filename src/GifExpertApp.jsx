import { useState } from 'react';
import { GifGrid , AddCategory } from './CMP'
export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Happy']);
    const onNewCategory = ( categoryValue ) => {
        if(categories.includes(categoryValue)) return;
        setCategories( category => [ categoryValue, ...category ] )
    }
    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onNewCategory = { onNewCategory }/>

            {
                categories.map(
                    category => (
                        <GifGrid key={category} category={category}/>
                    )
                )
            }        
        </>
    )
}
// api.giphy.com/v1/gifs/search?api_key=6fa0CYEvIeX1JrwhHnomwKeRodiqEKoe&q=valorant
