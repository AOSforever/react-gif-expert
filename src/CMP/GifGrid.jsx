import { useState,useEffect } from 'react';
// import { useEffect } from "react";
import { getGifs } from "../JS/getGifs";
import { GifItem } from './GifItem';
import { useFetchGifs } from '../HOOKS/useFetchGifs';

export const GifGrid = ({ category }) => {

  const { images, loading } = useFetchGifs( category );
  
  return (
    <>
        {
          loading && <h2>Cargando...</h2>
        }

        <h3>{ category }</h3>
        <div className='card-grid'>
          {
            images.map( image => (
              <GifItem { ...image } key={image.id}/>
            ))
          }
        </div>
    </>
  )
}
