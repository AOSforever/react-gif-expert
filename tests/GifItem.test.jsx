import { screen, render} from '@testing-library/react'
import { GifItem } from '../src/CMP';

describe('Pruebas en <GifItem/>', () => { 
    const url = 'https://jonhson.com/image.gif';
    const title = 'Imagen de asistencia';
    test('Debe hacer Match con el Snapshot', () => {
        const { container } = render(<GifItem url={ url }/>);
        expect( container ).toMatchSnapshot();
    })    

    test('La imagen debe de cargar correctamente',()=>{
        render(<GifItem url={url} title={ title } />);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })
    
    test('El titulo debe de cargar correctamente', () => {
        render(<GifItem url={url} title={ title } />);
        expect( screen.getByTestId('ThisTitle').innerHTML ).toBe(title)
    })
})
