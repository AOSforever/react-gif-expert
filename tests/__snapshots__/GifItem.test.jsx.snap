// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer Match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://jonhson.com/image.gif"
    />
    <p
      data-testid="ThisTitle"
    />
  </div>
</div>
`;
